src/jit/emitxarch.cpp

```diff
@@ -8008,28 +8183,57 @@ BYTE* emitter::emitOutputAM(BYTE* dst, instrDesc* id, code_t code, CnsVal* addc)
                     noway_assert((int)dsp == dsp);
 
                     // This requires, specifying a SIB byte after ModRM byte.
-                    dst += emitOutputWord(dst, code | 0x0400);
+                    if (Is4ByteAVXInstruction(ins))
+                    {
+                        dst += emitOutputByte(dst, code | 0x04);
+                    }
+                    else
+                    {
+                        dst += emitOutputWord(dst, code | 0x0400);
+                    }
                     dst += emitOutputByte(dst, 0x25);
 #endif //_TARGET_AMD64_
                     dst += emitOutputLong(dst, dsp);
                 }
                 break;
 
             case REG_EBP:
-                // Does the offset fit in a byte?
-                if (dspInByte)
+                if (Is4ByteAVXInstruction(ins))
                 {
-                    dst += emitOutputWord(dst, code | 0x4500);
-                    dst += emitOutputByte(dst, dsp);
+                    // Does the offset fit in a byte?
+                    if (dspInByte)
```

Comment:
This sequence is duplicated enough that it would be beneficial to pull it out into a method, I think.

Reply:
Logged https://github.com/dotnet/coreclr/issues/15879

Reply:
Thanks!